process/sync/storageBootstrap/shardStorageBootstrapper_test.go

```diff
@@ -252,3 +259,129 @@ func TestShardStorageBootstrapper_GetCrossNotarizedHeaderNonceShouldWork(t *test
 	assert.Nil(t, err)
 	assert.Equal(t, uint64(2), nonce)
 }
+
+func TestShardStorageBootstrapper_applyCrossNotarizedHeaders(t *testing.T) {
+	t.Parallel()
+
+	t.Run("missing current meta should error", func(t *testing.T) {
+		addedCrossNotarized := make(map[string]data.HeaderHandler)
+		addedTrackedHeaders := make(map[string]data.HeaderHandler)
+
+		bootstrapper, _, currentMeta, crossNotarizedHeaders := setupForApplyCrossNotarizedHeadersTests(addedCrossNotarized, addedTrackedHeaders)
+		_ = bootstrapper.store.GetStorer(dataRetriever.MetaBlockUnit).Remove(currentMeta.hash)
+
+		err := bootstrapper.applyCrossNotarizedHeaders(crossNotarizedHeaders)
+		assert.NotNil(t, err)
+		assert.True(t, strings.Contains(err.Error(), "missing header : GetMarshalizedHeaderFromStorage"))
+	})
+	t.Run("missing prev header should still work", func(t *testing.T) {
+		addedCrossNotarized := make(map[string]data.HeaderHandler)
+		addedTrackedHeaders := make(map[string]data.HeaderHandler)
+
+		bootstrapper, prevMeta, currentMeta, crossNotarizedHeaders := setupForApplyCrossNotarizedHeadersTests(addedCrossNotarized, addedTrackedHeaders)
+		_ = bootstrapper.store.GetStorer(dataRetriever.MetaBlockUnit).Remove(prevMeta.hash)
+
+		err := bootstrapper.applyCrossNotarizedHeaders(crossNotarizedHeaders)
+		assert.Nil(t, err)
+
+		assert.Equal(t, 1, len(addedCrossNotarized))
+		assert.Equal(t, 1, len(addedTrackedHeaders))
+
+		assert.Equal(t, currentMeta.metablock, addedCrossNotarized[string(currentMeta.hash)])
+		assert.Equal(t, currentMeta.metablock, addedTrackedHeaders[string(currentMeta.hash)])
+	})
+	t.Run("should work", func(t *testing.T) {
+		addedCrossNotarized := make(map[string]data.HeaderHandler)
+		addedTrackedHeaders := make(map[string]data.HeaderHandler)
+
+		bootstrapper, prevMeta, currentMeta, crossNotarizedHeaders := setupForApplyCrossNotarizedHeadersTests(addedCrossNotarized, addedTrackedHeaders)
+
+		err := bootstrapper.applyCrossNotarizedHeaders(crossNotarizedHeaders)
+		assert.Nil(t, err)
+
+		assert.Equal(t, 2, len(addedCrossNotarized))
+		assert.Equal(t, 2, len(addedTrackedHeaders))
+
+		assert.Equal(t, currentMeta.metablock, addedCrossNotarized[string(currentMeta.hash)])
+		assert.Equal(t, currentMeta.metablock, addedTrackedHeaders[string(currentMeta.hash)])
+
+		assert.Equal(t, prevMeta.metablock, addedCrossNotarized[string(prevMeta.hash)])
+		assert.Equal(t, prevMeta.metablock, addedTrackedHeaders[string(prevMeta.hash)])
+	})
+}
+
+func setupForApplyCrossNotarizedHeadersTests(
+	addedCrossNotarized map[string]data.HeaderHandler,
```

Comment:
`addedCrossNotarizedHeaders` / `crossNotarizedHeadersAccumulator` (optional renaming) - and above / below, if the second option is applied.

Reply:
renamed to `crossNotarizedHeadersAccumulator` and `trackedHeadersAccumulator`